jar-service/src/main/java/com/sixsq/slipstream/run/RuntimeParameterResource.java

```diff
@@ -169,23 +172,64 @@ private String extractValueFromEntity(Representation entity) {
 		}
 	}
 
+	private States attemptChangeGlobalState(String toState) {
+		States toNewState = null;
+		try {
+			toNewState = States.valueOf(toState);
+		} catch (IllegalArgumentException e) {
+			throwClientBadRequest("Requested transition to unknown state: " + toState);
+		}
+		States newState = null;
+		// Jumping between states with an abort flag set on the run is not a good practice.
+		if (isAbortSet()) {
+			throwClientBadRequest("For any state transition via API abort should be cleared first.");
+		} else {
+			Run run = Run.loadFromUuid(getUuid());
+			States currentState = run.getState();
+			if (run.isMutable() && States.Ready == currentState && States.Provisioning == toNewState) {
+				newState = attemptChangeGlobalStateToProvisioning();
+			} else {
+				throwClientBadRequest(String.format(
+				        "Via API state can be advanced only on a mutable run and only from %s to %s", States.Ready,
+				        States.Provisioning));
+			}
+		}
+		return newState;
+	}
+
+	private States attemptChangeGlobalStateToProvisioning() {
+		StateMachine sm = createStateMachine();
+		String fromToState = String.format("from %s to %s", States.Ready, States.Provisioning);
+		try {
+			sm.tryAdvanceToProvisionning();
+		} catch (InvalidStateException e) {
+			e.printStackTrace();
+			throwClientBadRequest(String.format("Failed to advance state %s: %s", fromToState, e.getMessage()));
+		} catch (CannotAdvanceFromTerminalStateException e) {
+			e.printStackTrace();
+			throwClientBadRequest(String.format("Failed to advance state %s: %s", fromToState, e.getMessage()));
+		}
+		States newState = sm.getState();
+		if (States.Provisioning != newState) {
+			throwServerError(String.format("Failed to advance state %s: requested doesn't match reached %s.",
+					fromToState, newState));
+		}
+		return newState;
+	}
+
 	@Post
-	public void completeCurrentNodeState(Representation entity) {
+	public void completeCurrentNodeStateOrChangeGlobalState(Representation entity) {
 
 		String nodeName = runtimeParameter.getNodeName();
-		States newState = null;
-
+		States newState;
```

Comment:
Does this change come from an hint of Intellij ? 
I would suggest the follwing:
```java
States newState = attemptCompleteCurrentNodeState(nodeName);
```